### DISCUSSION TITLE
Dictionaries and bonds

### DISCUSSION THREAD

**Discussion by Dan-Burns** [BradyAJohnston/MolecularNodes#310] [Ideas]
I'm inexperienced with blender but I want to get involved.  If  MOL_bonds_find is expensive and/or finds inaccurate bonds, can the bond data from the MDAnalysis universe be used?  Perhaps the spreadsheet can hold 5 or more new columns (bond1,bond2,....etc) and the columns contain the indices of atoms that the given row's atom is bonded to?  

Also I saw some comments about using dictionaries.  Perhaps a MolecularNodes python class can be used with an MDAnalysis universe and allow selections and representation keywords to be exported. Then some kind of import node can take all this info in and generate representations for specific selections.

Also, I'm assuming the rotation matrix utility can reorient the molecule in the blender reference frame - this would be nice to generate in MDAnalysis and import in a similar manner.

Thank you,
Dan

**Comment by BradyAJohnston** [comment]
Hi @Dan-Burns, thanks for the interest in getting involved!

In terms of bonds, if bond information is available inside an MDA topology, then it should automatically be imported in to the structure. With the resulting model that is created, there are vertices for each atom and edges between those vertices that represent the bonds.  We don't need to specially store bond indices on the point domain (atoms), as we can just create the edges which represent the bonds. 

If there is a specific topology file that is being read by MDA which isn't importing bonds as it should, please open an issue and we can discuss / troubleshoot there.

Currently when imported, the MDA Universe is discarded after topology and trajectory are imported, and everything is handled inside of Geometry Nodes. There is a PR #287 which maintains the connection to the underlying MDA session, allowing for updating selections and manipulation of the universe _inside_ of the blender session, which could potentially enable some of what you are suggesting with the export of groups. This PR is almost ready to merge for the MVP for maintaining a connection to the underlying session, but there is lots of further potential development in terms of improving the interface of Blender and the MDA universe, which I welcome any and all suggestions & contributions.
